### DISCUSSION TITLE
Timing of vnet creation and subnet requests

### DISCUSSION THREAD

**Discussion by positivism1** [Azure/ipam#157] [General]
Hi There! 

I am creating a new vnet via api and then immediately requesting the next available subnet via api (NextAvailableSubnet) feeding in the vnet id, obviously this fails because IPAM hasn't picked up the vnet yet. How long do I need to wait for this to happen? Is there an API I can use to check if its happened?

Also having a similiar problem with creating the subnet from NextAvailableSubnet, then immediately running NextAvailableSubnet again and getting the same subnet.

Currently for both of these putting a Start-Sleep -Seconds 360 solves this problem.

**Comment by DCMattyG** [comment]
Hi @positivism1, the delay that you're experiencing isn't really a delay in the IPAM tool, however we use Azure Resource Graph to gather details about resources in Azure. I believe the goal is to have 95% of resources in ARG within 1 minute of creation, but some resources can take a bit longer.

Here's the official docs on the topic:

https://learn.microsoft.com/en-us/azure/governance/resource-graph/how-to/get-resource-changes?tabs=azure-cli#understand-change-event-properties

In my testing, Virtual Networks are generally available in ARG in around 1 minute or less. Hope that helps!

**Reply by positivism1** [reply]
Thank you @DCMattyG dropped to 100 seconds and it works fine too.
